### PR TITLE
HealthDataRefactoring

### PR DESCRIPTION
**Pull request by prafulVaishnav**
1. Checking with testEnvironment so as to avoid collision with other unittest cases.
2. Making the preference name more clear
3. Using Date.now() instead of new Date().getTime()
4. Remove separate object to store userExtensions instead using ExtensionManager.extensions
5. Change function name to be more clear

### CODE REVIEW

File: src/extensions/default/HealthData/unittests.js [hunk 1]
```diff
@@ -65,14 +65,14 @@ define(function (require, exports, module) {
             });
 
             it("should send data to server", function () {
-                PreferencesManager.setViewState("lastTimeSendHealthData", (new Date()).getTime() - ONE_DAY);
+                PreferencesManager.setViewState("lastTimeSentHealthData", (new Date()).getTime() - ONE_DAY);
```
Comment (MarcelGerber) [inline src/extensions/default/HealthData/unittests.js]: You can use `Date.now()` here too (and below)

File: src/extensibility/ExtensionManager.js [hunk 2]
```diff
@@ -101,21 +101,9 @@ define(function (require, exports, module) {
...
-     *         status: the current status, one of the status constants above
-     */
-    var userExtensions = {};
-    
-    /**
-     * Registry object container  
+     * Check if we have updated registry object  
      */
-    var registryObject = {};
+    var isRegistryObjectUpdated = false;
```
Comment (prksingh) [inline src/extensibility/ExtensionManager.js]: Do you need this variable? You are not changing it anywhere

File: src/extensibility/ExtensionManager.js [hunk 3]
```diff
@@ -822,9 +794,8 @@ define(function (require, exports, module) {
     exports.updateExtensions        = updateExtensions;
     exports.getAvailableUpdates     = getAvailableUpdates;
     exports.cleanAvailableUpdates   = cleanAvailableUpdates;
-
-    exports.registryObject          = registryObject;
-
+    exports.isRegistryObjectUpdated = isRegistryObjectUpdated;
```
Comment (prksingh) [inline src/extensibility/ExtensionManager.js]: I think you can just set `exports.isRegistryObjectUpdated` to false and remove `isRegistryObjectUpdated` 

or you could make `isRegistryObjectUpdated` a method and then keep a private vairable you update in `downloadRegistry` and return true from `isRegistryObjectUpdated` when that variable is set.

**Comment by prksingh** [comment]
Looks good. Merging. Thanks!

**Pull request merged**